base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1231,7 +1249,10 @@ fn store_new_block<T: BlockchainBackend>(
             total_utxo.unwrap(),
         ),
     );
-
+    if algo == PowAlgorithm::Monero {
+        // The unwrap should be fine. It should only be none under non monero headers
+        db.update_monero_seed_height(&monero_seed.unwrap(), height)?;
```

Comment:
The unwrap can also be solved with an `if let Some()`....
```suggestion
        txn.insert_monero_seed_height(&monero_seed.unwrap(), height)?;
```